### PR TITLE
Feature/show the book data of an individual user

### PR DESCRIPTION
**Pull request by AmishaBisht**
#2454 

I have made a page to show all the details of users, the books they read, wishlisted and borrowed.

I have performed a self-review of my own code.

### CODE REVIEW

**Comment by amisha296** [comment]
Screenshot (111)

**Comment by amisha296** [comment]
Screenshot (112)

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 1]
```diff
@@ -55,7 +55,17 @@
...
     @endif
+<<<<<<< HEAD
+    <div class="d-flex justify-content-start flex-wrap" id="books_table" 
+        data-books="{{ json_encode($books) }}" 
+        data-categories="{{ json_encode($categories) }}"
+        data-index-route="{{ route('books.index') }}" 
+        data-reader-index-route="books/reader" 
+        data-category-index-route="{{ route('books.category.index') }}"
+        data-logged-in-user="{{ json_encode(auth()->user()) }}">
+=======
```
[the remark this answers is not in the document]
Comment (amisha296) [reply resources/views/knowledgecafe/library/books/index.blade.php:66]: okay

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
+@extends('layouts.app')
+@section('content')<!doctype html>
+<html lang="en">
+  <head>
+    <!-- Required meta tags -->
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <!-- Bootstrap CSS -->
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:10]: Use css from assets

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  </head>
+  <body style="background-color: rgba(177, 170, 129, 0.15); font-family: Muli, sans-serif;">
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:13]: style to css

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
+@extends('layouts.app')
+@section('content')<!doctype html>
+<html lang="en">
+  <head>
+    <!-- Required meta tags -->
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <!-- Bootstrap CSS -->
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:9]: The head section already comes from  `@extends('layouts.app') @section('content')` from here

**Review by gautam071 — CHANGES_REQUESTED** [review]
left some comments

**Comment by gautam071** [comment]
@amisha296 please check the conflicts, also there are some changes with the `composer.lock` file please check that

**Comment by amisha296** [comment]
> @amisha296 please check the conflicts, also there are some changes with the `composer.lock` file please check that

Resolved the merge conflict, I am not getting any idea of composer.lock, I didn't make the changes in there!

File: app/Http/Controllers/UserBookController.php [hunk 5]
```diff
@@ -22,4 +23,10 @@ public function booksInWishlist()
 
         return $user->booksInWishlist;
     }
+    public function userBookDetails(User $user)
+    {
+        $books = (new BookServices)->getReaderDetails($user);
```
Comment (gautam071) [inline app/Http/Controllers/UserBookController.php:28]: What does this do?

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 6]
```diff
@@ -102,16 +109,18 @@
...
+                        </a>
                     </div>
                 </div>
 
                 <div v-if="book.borrowers && book.borrowers.length">
-                    <p class="mb-0 mt-1">Borrowed by</p>
-                    <div class="pl-0 pt-2 pb-3">
-                        <img v-for="borrower in book.borrowers" :src="borrower.avatar" :alt="borrower.name" :title="borrower.name" class="reader_image mr-2 rounded-circle" data-toggle="tooltip" data-placement="bottom">
+                    <p  class="mb-0 mt-1">Borrowed by</p>
+                    <div  class="pl-0 pt-2 pb-3">
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/index.blade.php:122]: Indentations
```suggestion
                    <p class="mb-0 mt-1">Borrowed by</p>
                    <div class="pl-0 pt-2 pb-3">
```

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 7]
```diff
@@ -126,7 +135,6 @@
                         <h2 class="badge badge-warning px-2 py-1 mr-1">@{{ book.wishers.find(user => user.id == loggedInUser.id) == null ? '' : 'Wishlisted' }}</h2>
                     </span>
                 </div>
-
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/index.blade.php:129]: Add line back

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
+@extends('layouts.app')
+@section('content')<!doctype html>
+    <body style="background-color: #b1aa8126; font-family: Muli, sans-serif;">
+      <div class="container">
+          <div class="mt-4 card">
+              <div class="card-header">
+                  <div class=" fs-1 my-3 mx-4 px-2"> User Details</div>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:7]: Make the heading according as discussed!

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+              </div>
+              <div class="card-body">
+                  <div class="fs-2 d-flex justify-content-between mx-4 px-2 align-items-end">{{$user->name}}</div>
+                  <hr class='bg-dark mx-4 pb-0.5'></div>
+                  <div class="mx-5">
+                      <div class="container mb-3">
+                          <div class="row">
+                            <div class="col-sm border px-0">
+                              <h4 class="card-header" style="color: #1B65D5">Wishlisted Books({{$books['wishlistBookCount']}})</h4>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:16]: Use CSS!

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+                              @foreach ($user->booksInWishlist as $list)
+                              <div class="px-3 py-2 border-bottom">
+                                  {{$list->title}}
+                              </div>
+                              @endforeach
+                            </div>
+                            <div class="col-sm border px-0">
+                              <h4 class="card-header" style="color: #0A6B51">Read Books({{$books['readBookCount']}})</h4>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:24]: Use CSS

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+                              @foreach ($books['readBooks'] as $readBook)
+                              <div class="px-3 py-2 border-bottom">
+                                  {{$readBook->title}}
+                              </div>
+                              @endforeach
+                            </div>
+                            <div class="col-sm border px-0">
+                              <h4 class="card-header"  style="color: #E32315">Borrowed Books({{$books['borrowedBookCount']}})</h4>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/user-books-details.blade.php:32]: Use CSS

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserBookController.php (hunk 5 above)
Comment (amisha296) [reply app/Http/Controllers/UserBookController.php:28]: It calls the getReaderDetails function from the service file.

File: resources/views/knowledgecafe/library/books/user-books-details.blade.php (hunk 11 above)
Comment (amisha296) [reply resources/views/knowledgecafe/library/books/user-books-details.blade.php]: Did you mean classes?

**Comment by gautam071** [comment]
@amisha296 Can you check the requested changes

**Comment by amisha296** [comment]
> @amisha296 Can you check the requested changes

@gautam071 where to make the changes?

**Comment by amisha296** [comment]
> @amisha296 Can you check the requested changes

@gautam071 What changes?

**Pull request closed without merging**